src/ray/core_worker/task_interface.h

```diff
@@ -89,6 +127,20 @@ class CoreWorkerTaskInterface {
  private:
   /// Reference to the parent CoreWorker instance.
   CoreWorker &core_worker_;
+
+ private:
+  /// Build the arguments for a task spec.
+  ///
+  /// \param[in] args Arguments of a task.
+  /// \return Arguments as required by task spec.
+  std::vector<std::shared_ptr<raylet::TaskArgument>> BuildTaskArguments(
+      const std::vector<TaskArg> &args);
+
+  /// Translate from WorkLanguage to Language type (required by taks spec).
+  ///
+  /// \param[in] language Language for a task.
+  /// \return a pair of <Status, Language>.
+  std::pair<Status, ::Language> ToTaskLanguage(WorkerLanguage language);
```

Comment:
I'm not sure if we should use pairs in this way. The way this is normally done is via an out argument:
```
Status ToTaskLanguage(WorkerLanguage language, ::Language* out)
```

In this case, we might even do `::Lanuage ToTaskLanguage(WorkerLanguage language)` and have a check failure if the Language cannot be converted since it is probably a bug in that case.

Reply:
Revised to use the suggestion function signature.